src/plugins/cft/api.js

```diff
@@ -0,0 +1,244 @@
+}
+
+/**
+ * @param dateFrom
+ * @returns {Promise.<Array>}
+ */
+async function fetchTransactions(dateFrom) {
+    let isFirstPage  = true;
+    let transactions = [];
+
+    const lastSyncTime = new Date(dateFrom).getTime();
+    const pagination   = {
+        limit:  20,
+        offset: 0,
+        total:  null,
+    };
+
+    console.log('last sync: ' + lastSyncTime + ' (' + dateFrom + ')');
+
+    while (isFirstPage || pagination.offset < pagination.total) {
+        const data = await fetchTransactionsInternal(pagination.limit, pagination.offset);
+
+        if (data.part.offset !== pagination.offset) { // банк ограничивает выборку
+            break;
+        }
+
+        if (isFirstPage) {
+            pagination.total = data.part.totalCount;
+            isFirstPage      = false;
+        }
+
+        const loadNext = data.data
+                             .filter((transaction) => transaction.itemType === 'OPERATION' && ('money' in transaction))
+                             .every((transaction) => {
+                                 const isActual = transaction.date > lastSyncTime;
+
+                                 isActual && transactions.push(transaction);
+
+                                 return isActual;
+                             });
+
+        if (loadNext) {
+            pagination.offset += pagination.limit;
+        } else {
+            break;
+        }
+    }
+
+    return transactions;
+}
+
+/**
+ * @param limit
+ * @param offset
+ * @returns {Promise.<Array>}
+ */
+async function fetchTransactionsInternal(limit, offset) {
+    const urlParams = {
+        'limit':  limit,
+        'offset': offset,
+    };
+
+    const response = await fetchJson(makeApiUrl("hst", urlParams), {
+        log:                 logFetches,
+        headers:             defaultHeaders,
+        sanitizeRequestLog:  {},
+        sanitizeResponseLog: {},
+    });
+
+    assertResponseSuccess(response, [
+        'OK',
+        'OK_SYNC',
+    ]);
+
+    return response.body;
+}
+
+/**
+ * @param response
+ * @param allowedStatuses
+ */
+const assertResponseSuccess = (response, allowedStatuses = ['OK']) => {
+    console.assert(
+        response.status === 200 && allowedStatuses.indexOf(response.body.status) > -1,
+        "non-successful response",
+        response
+    );
+};
+
+/**
+ * @param path
+ * @param params
+ * @returns {string}
+ */
+const makeApiUrl = (path, params = {}) => {
+    params['rid'] = generateHash();
+
+    const query = Object.keys(params)
+                        .map((key) => {
+                            return [
+                                key,
+                                params[key]
+                            ].map(encodeURIComponent).join("=")
+                        })
+                        .join("&");
+
+    return apiUri + path + '?' + query;
+};
+
+/**
+ * @returns {string}
+ */
+const generateHash = () => {
+    return '0123456789abcdf'.repeat(5).split('').sort(() => {
```

Comment:
you missed `e`
I'd better generate 13 random ints in 0..15 range and then format each as hexadecimal instead of sorting by rand